app.js

```diff
@@ -1,15 +1,30 @@
 const express = require('express');
 const app = express();
-const { getTopics, getAllEndpoints } = require('./controlers/controlers');
+const {
+  getTopics,
+  getAllEndpoints,
+  getArticleById,
+} = require('./controlers/controlers');
 
 app.use(express.json());
 
 app.get('/api/topics', getTopics);
 app.get('/api', getAllEndpoints);
-
+app.get('/api/articles/:article_id', getArticleById);
+app.use((err, req, res, next) => {
+  if (err.code === '22P02') {
+    res.status(400).send({ msg: 'Bad request' });
+  } else {
+    next(err);
+  }
+});
```

Comment:
Good work implementing this error handling middleware for the specific PSQL error code of an invalid input, then invoking next to move onto the next middleware chain.